lib/internal/bootstrap/loaders.js

```diff
@@ -96,6 +96,33 @@ const internalBindingWhitelist = new SafeSet([
   'zlib'
 ]);
 
+{
+  const { setPrepareStackTraceCallback } = getInternalBinding('errors');
+  const MainContextError = Error;
+  const ErrorToString = Error.prototype.toString;
+  setPrepareStackTraceCallback((globalThis, error, trace) => {
+    // Polyfill of V8's Error.prepareStackTrace API.
+    // https://crbug.com/v8/7848
+    // TODO(devsnek): evaluate if this can be eventually deprecated/removed
+    // `globalThis` is the global that contains the constructor which
+    // created `error`.
+    if (typeof globalThis.Error.prepareStackTrace === 'function') {
+      return globalThis.Error.prepareStackTrace(error, trace);
+    }
+    // We still have legacy usage that depends on the main context's
+    // `Error` being used, even when the error is from a different context.
+    if (typeof MainContextError.prepareStackTrace === 'function') {
```

Comment:
This actually seems incorrect? But I guess we'll have to leave it to a semver-major patch to remove this.

Reply:
yeah it's not ideal, but it could be very breaking to the ecosystem to remove it, removing it already broke node itself in v8's upstream tests